<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8">
<title>OMNeT++ Simulation Library: cTerminationException Class Reference</title>
<link href="doxygen.css" rel="stylesheet" type="text/css">
<link href="tabs.css" rel="stylesheet" type="text/css">
</head><body>
<!-- Generated by Doxygen 1.5.5 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="main.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
</div>
<div class="contents">
<h1>cTerminationException Class Reference<br>
<small>
[<a class="el" href="group__Internals.html">Internal classes</a>]</small>
</h1><!-- doxytag: class="cTerminationException" --><!-- doxytag: inherits="cException" --><code>#include &lt;<a class="el" href="cexception_8h-source.html">cexception.h</a>&gt;</code>
<p>
<div class="dynheader">
Inheritance diagram for cTerminationException:</div>
<div class="dynsection">

<p><center><img src="classcTerminationException.png" usemap="#cTerminationException_map" border="0" alt=""></center>
<map name="cTerminationException_map">
<area href="classcException.html" alt="cException" shape="rect" coords="0,0,139,24">
</map>
</div>

<p>
<a href="classcTerminationException-members.html">List of all members.</a><hr><a name="_details"></a><h2>Detailed Description</h2>
Thrown when the simulation is completed. 
<p>
For example, <a class="el" href="classcSimpleModule.html#8713bf9b4d4970fcfe94995dcb42ce9e" title="Causes the whole simulation to stop.">cSimpleModule::endSimulation()</a> throws this exception. Statistics object may also throw this exception to signal that accuracy of simulation results has reached the desired level. <table border="0" cellpadding="0" cellspacing="0">
<tr><td></td></tr>
<tr><td colspan="2"><br><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcTerminationException.html#42b56576f08f95583027f1ff96709e65">cTerminationException</a> (OppErrorCode errcode,...)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcTerminationException.html#9c5105c443602a4bf5a311e00992caa9">cTerminationException</a> (const char *msg,...)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcTerminationException.html#66a35ea486cf023a8bab4aa620b68f41">cTerminationException</a> (const <a class="el" href="classcTerminationException.html">cTerminationException</a> &amp;e)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">virtual <a class="el" href="classcTerminationException.html">cTerminationException</a> *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcTerminationException.html#3cc34be98273933d33c17328b9c0f5e6">dup</a> () const </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">virtual bool&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcTerminationException.html#800ea3ac1413f70600cb9118eb7c1003">isError</a> () const </td></tr>

</table>
<hr><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" name="42b56576f08f95583027f1ff96709e65"></a><!-- doxytag: member="cTerminationException::cTerminationException" ref="42b56576f08f95583027f1ff96709e65" args="(OppErrorCode errcode,...)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">cTerminationException::cTerminationException           </td>
          <td>(</td>
          <td class="paramtype">OppErrorCode&nbsp;</td>
          <td class="paramname"> <em>errcode</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&nbsp;</td>
          <td class="paramname"> <em>...</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td width="100%"></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Error is identified by an error code, and the message comes from a string table. 
<p>
The error string may expect printf-like arguments (s, d) which also have to be passed to the constructor. 
</div>
</div><p>
<a class="anchor" name="9c5105c443602a4bf5a311e00992caa9"></a><!-- doxytag: member="cTerminationException::cTerminationException" ref="9c5105c443602a4bf5a311e00992caa9" args="(const char *msg,...)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">cTerminationException::cTerminationException           </td>
          <td>(</td>
          <td class="paramtype">const char *&nbsp;</td>
          <td class="paramname"> <em>msg</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&nbsp;</td>
          <td class="paramname"> <em>...</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td width="100%"></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
To be called like printf(). 
<p>
The error code is set to eCUSTOM. 
</div>
</div><p>
<a class="anchor" name="66a35ea486cf023a8bab4aa620b68f41"></a><!-- doxytag: member="cTerminationException::cTerminationException" ref="66a35ea486cf023a8bab4aa620b68f41" args="(const cTerminationException &amp;e)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">cTerminationException::cTerminationException           </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classcTerminationException.html">cTerminationException</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>e</em>          </td>
          <td>&nbsp;)&nbsp;</td>
          <td width="100%"><code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
We unfortunately need to copy exception objects when handing them back from an activity(). 
<p>

</div>
</div><p>
<hr><h2>Member Function Documentation</h2>
<a class="anchor" name="3cc34be98273933d33c17328b9c0f5e6"></a><!-- doxytag: member="cTerminationException::dup" ref="3cc34be98273933d33c17328b9c0f5e6" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">virtual <a class="el" href="classcTerminationException.html">cTerminationException</a>* cTerminationException::dup           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td width="100%"> const<code> [inline, virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Virtual copy constructor. 
<p>
We unfortunately need to copy exception objects when handing them back from an activity(). 
<p>Reimplemented from <a class="el" href="classcException.html#7aae9b928c595645f2918a897eb14c2b">cException</a>.</p>

</div>
</div><p>
<a class="anchor" name="800ea3ac1413f70600cb9118eb7c1003"></a><!-- doxytag: member="cTerminationException::isError" ref="800ea3ac1413f70600cb9118eb7c1003" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">virtual bool cTerminationException::isError           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td width="100%"> const<code> [inline, virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Termination exceptions are generally not errors, but messages like "Simulation completed". 
<p>

<p>Reimplemented from <a class="el" href="classcException.html#e94e8963aaae31fc8609986128d38d52">cException</a>.</p>

</div>
</div><p>
<hr>The documentation for this class was generated from the following file:<ul>
<li><a class="el" href="cexception_8h-source.html">cexception.h</a></ul>
</div>
<hr size="1"><address style="text-align: right;"><small>Generated on Tue Mar 27 11:22:13 2012 for OMNeT++ Simulation Library by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img src="doxygen.png" alt="doxygen" align="middle" border="0"></a> 1.5.5 </small></address>
</body>
</html>
